content/blog/hpe-morpheus-enterprise-case-study-fix-network-pool-links-via-a-custom-task-plugin.md

```diff
@@ -0,0 +1,158 @@
+
+![Network with IP Pool](/img/morphblog_linknetwork_network_with_pool.png)
+
+Should we provision a VM instance into this network, a host record entry will be created in the IP pool:
+
+![Instance provisioned](/img/morphblog_linknetwork_instance_provisioned.png)
+
+![Host record created](/img/morphblog_linknetwork_host_record.png)
+
+By querying the host record via the REST API, we can see the link back to the ***ComputeServer*** within the instance:
+
+![Host record via REST API](/img/morphblog_linknetwork_api_host_record.png)
+
+## Provision without IP pool
+
+To illustrate the broken reference issue, we start off by removing the network-to-IP-pool association:
+
+![Network without pool](/img/morphblog_linknetwork_network_without_pool.png)
+
+Then provision a VM to the network:
+
+![Provision VM to network](/img/morphblog_linknetwork_provision_vm.png)
+
+![Provisioned VM to network](/img/morphblog_linknetwork_provisioned_vm.png)
+
+Next, we create a host entry manually within the pool:
+
+![New Pool IP](/img/morphblog_linknetworks_manual_host_record.png)
+
+Finally, we add the IP pool back onto the network:
+
+![Network pool re-added onto network](/img/morphblog_linknetwork_network_pool_readded.png)
+
+This time, we can see that the ***refId*** property is NOT populated on the ***ComputeServer***, despite the matching host name on the VM:
+
+![API host record with no link](/img/morphblog_linknetwork_api_host_record_nolink.png)
+
+## Compiling and loading the Custom Task Plugin
+
+Download or clone the plugin repository from <https://github.com/neilvrhpe/link-network-hosts>.
+Open the project directory and compile with the relevant ***gradlew*** (Linux) or ***gradlew.bat*** (Windows) script using the ***shadowJar*** argument:
+
+![Compile the plugin jar file](/img/morphblog_linknetworks_compile_jar.png)
+
+The compiled ***.jar*** file will be found in the ***build/libs*** subdirectory:
+
+![Compiled jar file in build/libs subdirectory](/img/morphblog_linknetworks_compiled_jar.png)
+
+Upload the ***.jar*** file to the ***Administration > Integrations > Plugins > Add*** dialog. The plugin should appear in the list as shown below:
+
+![Plugin uploaded and shown in list](/img/morphblog_linknetworks_plugin_uploaded.png)
+
+## Setup the Custom Task Workflow
+
+Uploading the plugin in the previous step introduced a new ***TaskType*** to the HPE Morpheus Enterprise appliance. This can be seen under the edit dialog of the uploaded plugin:
+
+![New task type introduced](/img/morphblog_linknetworks_new_tasktype.png)
+
+To use this new custom task type in the UI, we will provide an ***OptionSource*** input. We create the corresponding ***OptionList*** under ***Library > Options > Option Lists > Add***. The type is ***Plugin*** and the option list will be ***Link Network Hosts: getNetworkPools***, as provided by the plugin that we uploaded in the previous step. Provide 'ChooseNetworkPool' as the name, so that it matches the name in the next step:
```

Comment:
Recommended rewrite:

"To use this new custom task type in the UI, you must provide an OptionSource input. Create the corresponding OptionList..."

Later in the paragraph...

"as provided by the plugin that was uploaded in the previous step."